src/Chapter2.hs

```diff
@@ -731,7 +775,10 @@ the list with only those lists that contain a passed element.
 
 🕯 HINT: Use the 'elem' function to check whether an element belongs to a list
 -}
-contains = error "contains: Not implemented!"
+contains :: Int -> [[Int]] -> [[Int]]
+contains _ []       = []
+contains c (x : xs) = if c `elem` x then x : rest else rest
+  where rest = contains c xs
```

Comment:
```haskell
contains :: Int -> [[Int]] -> [[Int]]
contains c = filter $ elem c
```
Like this?

Reply:
Yep, almost. We are not covering function application and composition (`$` and `.`), but if you would like to use it you would need to be more careful. As you have done eta-reduction already on this function, you can't use `$` to the `elem c`.

The best option is to app parenthesis:

```haskell
contains :: Int -> [[Int]] -> [[Int]]
contains c = filter (elem c)
```

As without eta-reduction it would look like this:
```haskell
contains :: Int -> [[Int]] -> [[Int]]
contains c xs = filter (elem c) xs
```

Reply:
Understood now, thank you!